### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/linkerd2/commit/aaf440489e0ec77b49f56a4bd52a635bf250e293#commitcomment-41423663

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/linkerd2/commit/5b82c6c5ca8f9f0acf6149b554ee13c215f6082d

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -31,4 +31,4 @@ indent_size = 2
 
 [*.proto]
 indent_size = 2
-indent_stlye = space
+indent_style = space
```
Comment (jsoref) [inline .editorconfig:34]: This repository is the only repo that appears to use this spelling

File: README.md [hunk 2]
```diff
@@ -65,10 +65,10 @@ usually take place the last Wednesday of the month at 9am Pacific / 4pm UTC.
 
 We're a friendly group, so please feel free to join us!
 
-## Code of conduct
+## Code of Conduct
 
 This project is for everyone. We ask that our users and contributors take a few
-minutes to review our [code of conduct][coc].
+minutes to review our [Code of Conduct][CoC].
```
Comment (jsoref) [inline README.md:71]: I'm willing to drop any family of changes.

Do note that formally you follow https://github.com/cncf/foundation/blob/master/code-of-conduct.md which uses the case I'm suggesting.

File: bin/_test-helpers.sh [hunk 3]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-# Override CI's `set -e` default, so we can catch errors manually and display
+# Override CIs' `set -e` default, so we can catch errors manually and display
```
Comment (jsoref) [inline bin/_test-helpers.sh:3]: There are a number of ways to read what is intended here:

* {continuous integration} {possesses} {`set -e`}
* {continuous integration} {abbreviation marker} {`set -e`}
* {continuous integration} {plural marker} {`set -e`}

As this file is being used by multiple CI systems, I think the correct meaning is a cross between the first and the last, hence I'm moving the `'`.

File: cli/cmd/multicluster.go [hunk 4]
```diff
@@ -393,7 +393,7 @@ func newMulticlusterInstallCommand() *cobra.Command {
 	cmd.Flags().StringVar(&options.gatewayNginxImage, "gateway-nginx-image", options.gatewayNginxImage, "The nginx image to be used")
 	cmd.Flags().StringVar(&options.gatewayNginxVersion, "gateway-nginx-image-version", options.gatewayNginxVersion, "The version of nginx to be used")
 	cmd.Flags().StringVar(&options.dockerRegistry, "registry", options.dockerRegistry, "Docker registry to pull images from")
-	cmd.Flags().BoolVar(&options.remoteMirrorCredentials, "service-mirror-credentials", options.remoteMirrorCredentials, "Whether to install the service account which can be used by service mirror components in source clusters to discover exported servivces")
+	cmd.Flags().BoolVar(&options.remoteMirrorCredentials, "service-mirror-credentials", options.remoteMirrorCredentials, "Whether to install the service account which can be used by service mirror components in source clusters to discover exported services")
```
Comment (jsoref) [inline cli/cmd/multicluster.go:396]: Obviously this is a behavior change to anything that parses the output. I hope that isn't a problem.

File: cli/cmd/multicluster.go [hunk 5]
```diff
@@ -539,7 +539,7 @@ func newLinkCommand() *cobra.Command {
 
 			gatewayIdentity, ok := gateway.Annotations[k8s.GatewayIdentity]
 			if !ok || gatewayIdentity == "" {
-				return fmt.Errorf("Gatway %s.%s has no %s annotation", gateway.Name, gateway.Namespace, k8s.GatewayIdentity)
+				return fmt.Errorf("Gateway %s.%s has no %s annotation", gateway.Name, gateway.Namespace, k8s.GatewayIdentity)
```
Comment (jsoref) [inline cli/cmd/multicluster.go:542]: Similarly, this is a behavior change to anything that parses error output.

File: controller/api/destination/server_test.go [hunk 6]
```diff
@@ -164,7 +164,7 @@ func TestGet(t *testing.T) {
 		}
 
 		// We cancel the stream before even sending the request so that we don't
-		// need to call server.Get in a separate goroutine.  By pre-emptively
+		// need to call server.Get in a separate goroutine.  By preemptively
```
Comment (jsoref) [inline controller/api/destination/server_test.go:167]: This spelling is correct/canonical and already used in this corpus.

File: controller/cmd/service-mirror/cluster_watcher.go [hunk 7]
```diff
@@ -297,14 +297,14 @@ func (rcsw *RemoteClusterServiceWatcher) cleanupMirroredResources() error {
 		return RetryableError{[]error{innerErr}}
 	}
 
-	for _, endpt := range endpoints {
-		if err := rcsw.localAPIClient.Client.CoreV1().Endpoints(endpt.Namespace).Delete(endpt.Name, &metav1.DeleteOptions{}); err != nil {
+	for _, endpoint := range endpoints {
+		if err := rcsw.localAPIClient.Client.CoreV1().Endpoints(endpoint.Namespace).Delete(endpoint.Name, &metav1.DeleteOptions{}); err != nil {
```
Comment (jsoref) [inline controller/cmd/service-mirror/cluster_watcher.go:301]: I see no particular reason for this abbreviation (and the longer form is used elsewhere)

File: controller/cmd/service-mirror/events_formatting.go [hunk 8]
```diff
@@ -41,14 +41,14 @@ func formatService(svc *corev1.Service) string {
 	return fmt.Sprintf("Service: {name: %s, namespace: %s, annotations: [%s], labels [%s]}", svc.Name, svc.Namespace, formatMetadata(svc.Annotations), formatMetadata(svc.Labels))
 }
 
-func formatEndpoints(endp *corev1.Endpoints) string {
+func formatEndpoints(endpoints *corev1.Endpoints) string {
```
Comment (jsoref) [inline controller/cmd/service-mirror/events_formatting.go:44]: This was actually confusing, as one would expect it to not be a plural object based on its name...

File: controller/cmd/service-mirror/events_formatting.go [hunk 9]
```diff
@@ -9,12 +9,12 @@ import (
 )
 
 func formatAddresses(addresses []corev1.EndpointAddress) string {
-	var adrs []string
+	var addrs []string
```
Comment (jsoref) [inline controller/cmd/service-mirror/events_formatting.go:12]: This was an unusual abbreviation...

File: grafana/dashboards/multicluster.json [hunk 10]
```diff
@@ -441,7 +441,7 @@
                     "expr": "sum(irate(request_total{dst_target_gateway=\"$gateway\", dst_target_gateway!=\"\", dst_target_cluster=\"$cluster\", dst_target_cluster!=\"\", tls!=\"true\"}[30s])) by (dst_target_gateway)",
                     "format": "time_series",
                     "intervalFactor": 1,
-                    "legendFormat": "gatewy/{{dst_target_gateway}}",
+                    "legendFormat": "gateway/{{dst_target_gateway}}",
```
Comment (jsoref) [inline grafana/dashboards/multicluster.json:444]: I hope this isn't a problem

File: pkg/charts/charts.go [hunk 11]
```diff
@@ -24,8 +24,8 @@ type Chart struct {
 	Files     []*chartutil.BufferedFile
 }
 
-func (chart *Chart) render(partialsFiles []*chartutil.BufferedFile) (bytes.Buffer, error) {
-	if err := FilesReader(chart.Dir+"/", chart.Files); err != nil {
+func (c *Chart) render(partialsFiles []*chartutil.BufferedFile) (bytes.Buffer, error) {
+	if err := FilesReader(c.Dir+"/", c.Files); err != nil {
```
Comment (jsoref) [inline pkg/charts/charts.go:28]: A significant portion of this repository uses the single character notation, including files relatively close to this one.

Using that notation for this makes it possible to make the change below.

File: pkg/charts/charts.go [hunk 12]
```diff
@@ -34,31 +34,31 @@ func (chart *Chart) render(partialsFiles []*chartutil.BufferedFile) (bytes.Buffe
...
-			Namespace: chart.Namespace,
+			Namespace: c.Namespace,
 		},
 		KubeVersion: "",
 	}
 
-	chrtConfig := &helmChart.Config{Raw: string(chart.RawValues), Values: map[string]*helmChart.Value{}}
-	renderedTemplates, err := renderutil.Render(chrt, chrtConfig, renderOpts)
+	chartConfig := &helmChart.Config{Raw: string(c.RawValues), Values: map[string]*helmChart.Value{}}
+	renderedTemplates, err := renderutil.Render(chart, chartConfig, renderOpts)
```
Comment (jsoref) [inline pkg/charts/charts.go:54]: This is the easiest place to see the change as a whole.

File: pkg/inject/inject.go [hunk 13]
```diff
@@ -1116,7 +1116,7 @@ func (conf *ResourceConfig) InjectNamespace(annotations map[string]string) ([]by
 	return yaml.JSONToYAML(j)
 }
 
-//getFilteredJSON method performs JSON marshaling such that zero values of
+//getFilteredJSON method performs JSON marshalling such that zero values of
```
Comment (jsoref) [inline pkg/inject/inject.go:1119]: There are various opinions about `l`s in marshal*, the set I've picked seem to be somewhat dominant in general and in this repository.

File: pkg/issuercerts/issuercerts.go [hunk 14]
```diff
@@ -125,7 +125,7 @@ func CheckExpiringSoon(cert *x509.Certificate) error {
 // we have posed on the public key and signature algorithms
 func CheckCertAlgoRequirements(cert *x509.Certificate) error {
 	if cert.PublicKeyAlgorithm == x509.ECDSA {
-		// this si a safe cast here as wel know we are using ECDSA
+		// this si a safe cast here as well know we are using ECDSA
```
Comment (jsoref) [inline pkg/issuercerts/issuercerts.go:128]: `is` is misspelled. I can fix that as well. I'm not sure offhand why I missed that.

File: pkg/tls/ca_test.go [hunk 15]
```diff
@@ -27,21 +27,21 @@ func TestCaIssuesCertsWithCorrectExpiration(t *testing.T) {
 		desc                   string
 		validFrom              time.Time
 		issuerLifeTime         time.Duration
-		endCertLifeime         time.Duration
+		endCertLifetime         time.Duration
```
Comment (jsoref) [inline pkg/tls/ca_test.go:30]: I expect there's a go linter that will scream at me for this. Sorry.

File: web/app/js/components/util/TapUtils.jsx [hunk 16]
```diff
@@ -102,45 +102,45 @@ export const processNeighborData = (source, labels, resourceAgg, resourceType) =
     return resourceAgg;
   }
 
-  let neighb = {};
+  let neighbor = {};
```
Comment (jsoref) [inline web/app/js/components/util/TapUtils.jsx:105]: This abbreviation is odd. A lot of code in this repository and even in this file doesn't use abbreviations. Here, the abbreviation makes it harder to understand the intent.

Is it `neighbors` or `neighbor`? I'm not really sure.

File: pkg/issuercerts/issuercerts.go (hunk 14 above)
Comment (jsoref) [reply pkg/issuercerts/issuercerts.go:128]: reading it again, I think this change is wrong and that the correct word is `we` not `well`.

File: bin/_test-helpers.sh (hunk 3 above)
Comment (adleong) [reply bin/_test-helpers.sh:3]: I think this is intended to be singular possessive so I think "CI's" is correct.

File: cli/cmd/multicluster.go (hunk 4 above)
Comment (adleong) [reply cli/cmd/multicluster.go:396]: This is just the flag description so I don't think anything is parsing this output.

File: pkg/tls/codec.go [hunk 17]
```diff
@@ -105,7 +105,7 @@ func DecodePEMCertificates(txt string) (certs []*x509.Certificate, err error) {
 	return
 }
 
-// CertificatesToPool covererts a slice of certificates into a cert pool
+// CertificatesToPool coverts a slice of certificates into a cert pool
```
Comment (adleong) [inline pkg/tls/codec.go:108]: I think this should be "converts"

File: pkg/tls/ca.go [hunk 18]
```diff
@@ -157,7 +157,7 @@ func GenerateRootCAWithDefaults(name string) (*CA, error) {
 	return CreateRootCA(name, key, Validity{})
 }
 
-// GenerateCA generates a new intermdiary CA.
+// GenerateCA generates a new intermediate CA.
```
Comment (adleong) [inline pkg/tls/ca.go:160]: I think "intermediary" was intentional, but "intermediate" seems to be a more common term for this.  :+1:

Comment (jsoref) [reply pkg/tls/ca.go:160]: Yeah, that's exactly the thought process I took.

File: pkg/tls/codec.go (hunk 17 above)
Comment (jsoref) [reply pkg/tls/codec.go:108]: Indeed. I think I had that in mind at one point but then was careless in copying a correction.

File: pkg/profiles/profiles_test.go [hunk 19]
```diff
@@ -80,7 +80,7 @@ spec:
       pathRegex: /route-1`,
 		},
 		{
-			err: errors.New("failed to validate ServiceProfile: error unmarshaling JSON: while decoding JSON: json: unknown field \"foo\""),
+			err: errors.New("failed to validate ServiceProfile: error unmarshalling JSON: while decoding JSON: json: unknown field \"foo\""),
```
Comment (adleong) [inline pkg/profiles/profiles_test.go:83]: I think this error message actually comes from a dependency so we need to leave the "unmarshaling" spelling in this expectation.

Comment (jsoref) [reply pkg/profiles/profiles_test.go:83]: Yep. It took me a bit and I was starting to rework things to tolerate that, and then got interrupted.

It's unfortunate that the world is split on the spelling, but it won't heal anytime soon.

I'll have that tolerated w/ my next force-push.

File: grafana/dashboards/prometheus-benchmark.json [hunk 20]
```diff
@@ -1360,7 +1360,7 @@
           "step": 10
         },
         {
-          "expr": "rate(prometheus_tsdb_wal_truncate_duration_seconds_sum{job=\"prometheus\",instance=\"$Prometheus:9090\"}[1m]) / rate(prometheus_tsdb_wal_trunacte_duration_seconds_count{job=\"prometheus\",instance=\"$Prometheus:9090\"}[1m])",
+          "expr": "rate(prometheus_tsdb_wal_truncate_duration_seconds_sum{job=\"prometheus\",instance=\"$Prometheus:9090\"}[1m]) / rate(prometheus_tsdb_wal_truncate_duration_seconds_count{job=\"prometheus\",instance=\"$Prometheus:9090\"}[1m])",
```
Comment (jsoref) [inline grafana/dashboards/prometheus-benchmark.json:1363]: I meant to flag this. This seems to be a typo which doesn't appear anywhere else.

**Pull request merged**